R/tree.R

```diff
@@ -268,6 +268,22 @@ is_printable_env <- function(x) {
     )
 }
 
+tree_as_list <- function(x){
```

Comment:
What kind of objects can this take as input? I'm asking because the removal of all attributes (other than `names`) may have unexpected results. For example:

```R
x <- matrix(1:6, 2, 3)
attributes(x)
#> $dim
#> [1] 2 3

x <- data.frame(x=1:3, y=4:6)
attributes(x)
#> $names
#> [1] "x" "y"
#> 
#> $class
#> [1] "data.frame"
#> 
#> $row.names
#> [1] 1 2 3
```

Reply:
Stripping attributes seems like the right approach to me.

Reply:
It's only used to make sure we can properly iterate over the children.  We already pull off attributes before doing this if they are requested with `show_attributes = TRUE`.

Reply:
Can you add a comment about what kind of objects this function can take as input? That'll help future readers to understand what should be happening in this function.

Reply:
So I realized that I only called this function once. (I had wrapped `attributes(x)` in it, but since `attributes()` always returns a list I didn't need to there.) So I moved the function inline and streamlined it a bit. 
```r
...
    # Coerce current object to a plain list. This is necessary as some s3
    # classes override `[[` and return funky stuff like themselves (see s3 class
    # "package_version")
    children <- if (is_printable_env(x)) {
      # Environments are funky as they don't have names before conversion to list
      # but do after, so let them handle their conversion.
      # We use all.names = TRUE in an effort to fully explain the object
      as.list.environment(x, all.names = TRUE)
    } else {
      # By wiping all attributes except for the names we force the object to be
      # a plain list. This is inspired by the (now depreciated) rlang::as_list().
      attributes(x) <- list(names = names(x))
      as.list(x)
    }
...
```